### PR TITLE
ci: Build isolated docs to validate syntax

### PR DESCRIPTION
**Pull request by alexjuda**
# The problem

Reproducing docs build issues required running full docs build (the one that produces `docs.orquestra.io`). It was far removed from our development; the feedback cycle was long, and it was tedious to reproduce the issues.

# This PR's solution

* Adds a sphinx build configuration. It's based on the one used for full docs suite, but it produces docs for only this repo. The docs build will fail is any warnings is emitted. This allows catching .rst syntax issues.
* Adds a GH workflow that runs the docs build just to check against build issues.

The "docs" GH workflow will fail until https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/304 is merged.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

**Review by SebastianMorawiec — CHANGES_REQUESTED** [review]
Question: png and ico files belong to LFS. Do we have LFS support?

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    "sphinx.ext.graphviz",
+    "sphinxcontrib.youtube",
+    "sphinxcontrib.autoprogram",
+    "sphinx_copybutton",
+    "sphinxemoji.sphinxemoji",
+    "sphinx_click",
+]
+source_suffix = {
+    ".rst": "restructuredtext",
+    # '.txt': 'restructuredtext',
```
Comment (SebastianMorawiec) [inline docs/conf.py:43]: suggestion: Remove commented-out code. Its confusing

**Comment by alexjuda** [comment]
@SebastianMorawiec 

> Question: png and ico files belong to LFS. Do we have LFS support?

The included files weight 15kb and 24kb respectively. Configuring LFS for this would be an overkill.

**Comment by SebastianMorawiec** [comment]
> 

If no LFS is configured, first just make sure that those files are required. If they are, im fine with keeping them out of LFS

**Review by SebastianMorawiec — APPROVED** [review]

**Pull request merged**